packages/editor/src/browser/editor-widget.ts

```diff
@@ -32,6 +34,14 @@ export class EditorWidget extends BaseWidget implements SaveableSource, Navigata
                 this.selectionService.selection = this.editor;
             }
         }));
+        this.toDispose.push(this.labelProvider.onDidChange(async uri => {
+            const resourceUri = this.editor.getResourceUri();
+            if (resourceUri && resourceUri.toString() === uri.toString()) {
+                const icon = await this.labelProvider.getIcon(uri);
```

Comment:
I would be good to have protected `updateTitle` method which recomputes editor title based on the original `uri`. It should be called on editor construction as well on `onDidChange` event.